tests/test_parameters.py

```diff
@@ -0,0 +1,101 @@
+"""
+Testing for the ElfPy package modules
+"""
+
+# pylint: disable=too-many-instance-attributes
+# pylint: disable=too-many-locals
+# pylint: disable=attribute-defined-outside-init
+# pylint: disable=duplicate-code
+
+import unittest
+import logging
+
+import numpy as np
+import utils_for_tests as test_utils  # utilities for testing
+
+import elfpy.utils.outputs as output_utils  # utilities for file outputs
+
+
+class BaseParameterTest(unittest.TestCase):
+    """Generic Parameter Test class"""
+
+    def run_base_trade_test(
+        self,
+        agent_policies,
+        config_file="config/example_config.toml",
+        delete_logs=True,
+    ):
+        """Assigns member variables that are useful for many tests"""
+        output_utils.setup_logging(log_filename=".logging/test_parameters.log", log_level=logging.DEBUG)
+        override_dict = {
+            "num_trading_days": 3,  # sim 3 days to keep it fast for testing
+            "num_blocks_per_day": 3,  # 3 block a day, keep it fast for testing
+        }
+        simulator = test_utils.setup_simulation_entities(
+            config_file=config_file, override_dict=override_dict, agent_policies=agent_policies
+        )
+        simulator.run_simulation()
+        output_utils.close_logging(delete_logs=delete_logs)
+        return simulator
+
+    def run_custom_parameters_test(self, agent_policies, expected_result, delete_logs=True):
+        """Test custom parameters passed to agent creation"""
+        # create simulator with agent_policies
+        simulator = self.run_base_trade_test(agent_policies=agent_policies, delete_logs=delete_logs)
+        number_of_init_agents = 0  # count number of init agents so we can skip over them
+        for all_agent_index, agent in simulator.agents.items():  # loop over all agents
+            if agent.name == "init_lp":
+                number_of_init_agents += 1
+            else:  # only for custom agents, loop across them and check their parameters
+                custom_agent_index = all_agent_index - number_of_init_agents  # identify which custom agent we are on
+                expected_result_dict = expected_result[custom_agent_index]
+                for key, value in expected_result_dict.items():  # for each custom parameter to check
+                    np.testing.assert_almost_equal(
+                        getattr(agent, key),
```

Comment:
is there a reason why you would not expect these to be exactly equal?

Reply:
only python rounding. i think default is 1e-7, probably could be lower?

Reply:
wait no there's no rounding here